### PR TITLE
Alerting: golint fixes for alerting

### PR DESCRIPTION
**Pull request by bergquist**
I also enabled golinting for the alerting package.

Spring cleaning!

### CODE REVIEW

**Review by marefr — COMMENTED** [review]
Awesome! Left some comments

image

File: Makefile [hunk 1]
```diff
@@ -84,6 +84,11 @@ revive: scripts/go/bin/revive
 		-config ./scripts/go/configs/revive.toml \
 		$(GO_FILES)
 
+revive-alerting: scripts/go/bin/revive
```
[the remark this answers is not in the document]
Comment (bergquist) [reply Makefile]: the `revive` step uses the `/scripts/go/configs/revive.toml` config file to disable some checks. In this step we run all default linters.

File: pkg/services/alerting/conditions/reducer.go [hunk 2]
```diff
@@ -117,8 +117,8 @@ func (s *SimpleReducer) Reduce(series *tsdb.TimeSeries) null.Float {
 	return null.FloatFrom(value)
 }
 
-func NewSimpleReducer(typ string) *SimpleReducer {
-	return &SimpleReducer{Type: typ}
+func newSimpleReducer(typ string) *queryReducer {
```
[the remark this answers is not in the document]
Comment (bergquist) [reply pkg/services/alerting/conditions/reducer.go]: `type` is a reserved keyword. `t` might make more sense. Fix inc.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (marefr) [reply Makefile]: Got it :+1:

**Review by marefr — APPROVED** [review]
The force push made it hard to find what you changed since my last review, but think it seems fix :+1:

**Pull request merged**